### ISSUE TITLE
Refactor LibUsbDotNet project into 4 separate projects

### ISSUE THREAD

**Issue by nzsmith1** [LibUsbDotNet/LibUsbDotNet#11]
Refactor the LibUsbDotNet project into 4 separate projects to help re-architect the LibUsbDotNet solution and make it possible and easier to move to different repos.

The projects would basically be wrappers:
LibUsbDotNet.Core
-Includes all base classes, interfaces and common code between all remaining projects
-Build new interfaces as needed during refactoring
-Mostly what is in LibUsbDotNet and LibUsbDotNet.main namespaces

The remaining projects then based on @andrasfuchs original proposal in [Repos and refactoring #5](https://github.com/LibUsbDotNet/LibUsbDotNet/issues/5)

LibUsbDotNet.WinUSB
-Everything in the LibUsbDotNet.WinUsb namespace.

LibUsbDotNet.LibUsb
-Everything in the LibUsbDotNet.MonoLibUsb namespace

LibUsbDotNet.LibUsb32
-Everything in the LibUsbDotNet.LibUsb namespace

Steps to take to get there (each one of these will likely be its own PR

- [ ] Review all namespaces and implement any common interfaces
- [ ] Create LibUsbDotNet.Core project within solution
- [ ] Create remaining projects within solution and refactor namings based on new structure

I don't expect a lot of net new code here, just a better structure.  Once the structure is in place it will make it easier to implement missing dll methods in each wrapper as each will be self contained.

As we don't care about backwards comparability as defined in [Repos and refactoring #5](https://github.com/LibUsbDotNet/LibUsbDotNet/issues/5) this is a good opportunity to update the naming of these wrappers.

If anyone has any thoughts on this, please let me know.

**Comment by qmfrederik** [comment]
Looks good to me, I guess the first real test would be to try & separate the .Core part from everything else, if you already have that up & running happy to review a PR!
